src/nnet3/nnet-simple-component.cc

```diff
@@ -87,27 +87,32 @@ void PnormComponent::Write(std::ostream &os, bool binary) const {
 }
 
 
-void DropoutComponent::Init(int32 dim, BaseFloat dropout_proportion) {
+void DropoutComponent::Init(int32 dim, BaseFloat dropout_proportion,
+                            bool dropout_per_frame) {
   dropout_proportion_ = dropout_proportion;
+  dropout_per_frame_ = dropout_per_frame;
   dim_ = dim;
 }
 
 void DropoutComponent::InitFromConfig(ConfigLine *cfl) {
   int32 dim = 0;
   BaseFloat dropout_proportion = 0.0;
+  bool dropout_per_frame = true;
   bool ok = cfl->GetValue("dim", &dim) &&
     cfl->GetValue("dropout-proportion", &dropout_proportion);
+    // for this stage, dropout is hard coded in by-frame mode
   if (!ok || cfl->HasUnusedValues() || dim <= 0 ||
```

Comment:
where is the "dropout-per-frame" configuration value being read?
You should be doing
```
 cfl->GetValue("dropout-per-frame", &dropout_per_frame);
```
[don't check the return value of GetValue().]